src/ia-item-user-lists.ts

```diff
@@ -1,222 +1,231 @@
-    const createEventListener = (e: CustomEvent) => {
-      // TEMP: Set selected count for main button icon state
-      this.selectedCount += 1;
-
-      // eslint-disable-next-line no-console
-      console.log('createUserList listener', e.detail.created);
-
-      this.addMember(e.detail.created.id);
-
-      this.dispatchEvent(
-        new CustomEvent('closeDropdown', {
-          bubbles: true,
-          composed: true,
-        })
-      );
-    };
-    window.addEventListener(
-      'createUserList',
-      // eslint-disable-next-line no-undef
-      createEventListener as EventListener
-    );
-  }
+  // Events
 
-  connectedCallback(): void {
-    super.connectedCallback?.();
-    // Set userlist data
-    this.initUserLists();
+  async dropdownClicked(e: Event): Promise<void> {
+    e.preventDefault();
+    if (!this.dropdown.open) {
+      // get userlist data
+      await this.dataActionTask.run(['load']);
+      this.dropdown.open = true;
+    } else {
+      this.dropdown.open = false;
+    }
   }
 
-  private async addMember(listId: string): Promise<void> {
-    await this.userListsService.addMemberToList(listId, {
-      identifier: this.item,
-    });
-    await this.initUserLists();
+  // Event Handlers
+
+  // Listen for simple close Dropdown event from item-userlists
+  closeListener = (): void => {
+    this.dropdown.open = false;
+  };
+
+  // Listen for close and wait Dropdown event from item-userlists
+  selectListener = (): void => {
+    this.dropdown.open = false;
+    this.dataActionTask.run(['initial']);
+  };
+
+  // Listen for create List event from create-new-list
+  updateListener = (): void => {
+    this.dataActionTask.run(['load']);
+  };
+
+  // Lifecycle Methods
+
+  async firstUpdated(): Promise<void> {
+    // Give the browser a chance to paint
+    // eslint-disable-next-line no-promise-executor-return
+    await new Promise(r => setTimeout(r, 0));
+
+    // Setup event listeners
+    this.addEventListener('closeDropdown', this.closeListener);
+
+    this.addEventListener('selectDropdown', this.selectListener);
+
+    this.addEventListener('updateDropdown', this.updateListener);
+
+    this.dataActionTask.run(['load']);
   }
 
-  private async initUserLists(): Promise<void> {
-    // Load userlist data from API
-    // eslint-disable-next-line no-console
-    console.log('fetching userlist data for item', this.item);
+  // Tasks
+
+  /**
+   * Task managing all async data and UI updates
+   * @param action {DataAction} - Action to perform
+   * @param listId {string} - List ID
+   */
+  private dataActionTask = new Task(this, {
+    task: async ([action]) => {
+      if (!this.item || !this.userListsService) {
+        return initialState;
+      }
+      switch (action) {
+        case 'load':
+          return this.updateSelectedCount();
+        default:
+          return initialState;
+      }
+    },
+    args: () => [this.dataAction],
+    autoRun: false,
+  });
```

Comment:
Interesting, I haven't worked with `@lit/task` before. Is this basically an abstraction for long-running async calls that affect state? Seems pretty nifty!

Reply:
Yes, it takes care of a number of things: as well as preventing infinite loops!